policies/rules/security/entry.go

```diff
@@ -0,0 +1,750 @@
+
+import (
+	"encoding/xml"
+	"fmt"
+	"strings"
+
+	"github.com/PaloAltoNetworks/pango/filtering"
+	"github.com/PaloAltoNetworks/pango/generic"
+	"github.com/PaloAltoNetworks/pango/util"
+	"github.com/PaloAltoNetworks/pango/version"
+)
+
+var (
+	_ filtering.Fielder = &Entry{}
+)
+
+var (
+	Suffix = []string{"security", "rules"}
+)
+
+// Entry is the normalized object.
+//
+// Targets is a map where the key is the serial number of the target device and
+// the value is a list of specific vsys on that device.  The list of vsys is
+// nil if all vsys on that device should be included or if the device is a
+// virtual firewall (and thus only has vsys1).
+type Entry struct {
+	Name                            string
+	Uuid                            string
+	SourceZones                     []string // unordered
+	DestinationZones                []string // unordered
+	SourceAddresses                 []string // unordered
+	SourceUsers                     []string // unordered
+	DestinationAddresses            []string // unordered
+	Services                        []string // unordered
+	Categories                      []string // unordered
+	Applications                    []string // unordered
+	SourceDevices                   []string
+	DestinationDevices              []string
+	Schedule                        *string
+	Tags                            []string // ordered
+	NegateSource                    *bool
+	NegateDestination               *bool
+	Disabled                        *bool
+	Description                     *string
+	GroupTag                        *string
+	Action                          string
+	IcmpUnreachable                 *bool
+	Type                            *string
+	DisableServerResponseInspection *bool
+	LogSetting                      *string
+	LogStart                        *bool
+	LogEnd                          *bool
+	ProfileSettings                 *ProfileSettingsObject
+	Qos                             *QosObject
+	Targets                         map[string][]string
+	NegateTarget                    *bool
+	DisableInspect                  *bool // 10.2.4, 10.2.0? : disable-inspect
+
+	Misc map[string][]generic.Xml
+}
+
+type ProfileSettingsObject struct {
+	Groups   []string
+	Profiles *ProfilesObject
+}
+
+type ProfilesObject struct {
+	UrlFilteringProfiles     []string
+	DataFilteringProfiles    []string
+	FileBlockingProfiles     []string
+	WildfireAnalysisProfiles []string
+	AntiVirusProfiles        []string
+	AntiSpywareProfiles      []string
+	VulnerabilityProfiles    []string
+}
+
+type QosObject struct {
+	IpDscp                   *string
+	IpPrecedence             *string
+	FollowClientToServerFlow any
+}
+
+func (e *Entry) CopyMiscFrom(v *Entry) {
+	if v == nil || len(v.Misc) == 0 {
+		return
+	}
+
+	e.Misc = make(map[string][]generic.Xml)
+	for key := range v.Misc {
+		e.Misc[key] = append([]generic.Xml(nil), v.Misc[key]...)
+	}
+}
+
+func (e *Entry) Field(value string) (any, error) {
```

Comment:
I think the better approach is to actually go with reflect in this case, this whole logic we can put in few lines of code to generate what we need.

Reply:
This very well could use reflect, but as I brought up in our meeting this function should also be able to deal with other situations, such as indexing into a list of structs (neither address objects nor security rules has this situation, so there is no example of this yet).

My recommendation is to pursue reflect, but after all the other work is done.